### PR TITLE
[Improvement] Add Mail Lost Option as Replacement Reason

### PR DESCRIPTION
**Pull request by kj3moraes**
## Notion ticket link
[Add MAIL_LOST as a reason for replacement](https://www.notion.so/uwblueprintexecs/Add-MAIL_LOST-as-a-reason-for-replacement-a09d592f270c433b9a0e0ef8e1236385?pvs=4)

## Implementation description
* made a modification to the prisma schema to add MAIL_LOST as an option
* made a modification to the reason replacement form to add mail lost as a radio button 
* made a modification to the GQL schema to add MAIL_LOST as an option

## Notes
* I had to add a migration using `prisma migrate deploy` that manually added another migration. I don't know if this is supposed to be done because there was only an init migration before me. Please check that this is valid
* For LOST,STOLEN and OTHER there are additional fields for information. Will I need to add additional fields for MAIL_LOST 
* [Loom demoing it working](https://www.loom.com/share/f3b39c6a53584c9d800f3799dd925324?sid=d995e77d-0f31-48a1-ba9e-4825791b58e5)

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

**Comment by kj3moraes** [comment]
It says that the Prettier linting has failed but I just ran it on my machine and this is what I get 
CleanShot 2024-01-23 at 11 00 00

File: components/admin/requests/reason-for-replacement/Form.tsx
Comment (sherryhli) [inline components/admin/requests/reason-for-replacement/Form.tsx:65]: Just occurred to me that we need to discuss how to handle existing replacement applications that were created after #325, which mapped the `OTHER` enum to the displayed value of 'Mail Lost'. Once this is PR merged, any application using `OTHER` will show as 'Other' even if they were supposed to be 'Mail Lost'.

Let's discuss in Wednesday's meeting, can you please hold off on merging this PR (at least the frontend portion)?

**Review by sherryhli — COMMENTED** [review]
**Re: lint error**
Seems related to the changes in the yarn.lock file - reverting back to the current version in staging fixes it. See #342


> I had to add a migration using prisma migrate deploy that manually added another migration. I don't know if this is supposed to be done because there was only an init migration before me. Please check that this is valid

Migration looks good to me! For future reference the procedure is to run `prisma migrate dev` after any schema change.

> For LOST,STOLEN and OTHER there are additional fields for information. Will I need to add additional fields for MAIL_LOST

Do you mean fields like `lostTimestamp`, `stolenPoliceFileNumber`, etc. in the [schema](https://github.com/uwblueprint/richmond-centre-for-disability/blob/staging/prisma/schema.prisma#L315-L329)? Not necessary if that's what you're referring to.

**Review by ChinemeremChigbo — APPROVED** [review]

**Pull request merged**
